end_to_end_tests/golden-record-custom/custom_e2e/models/model_with_any_json_properties.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict, List, Union, cast
+
+import attr
+
+from ..models.model_with_any_json_properties_additional_property import ModelWithAnyJsonPropertiesAdditionalProperty
+from ..types import Unset
+
+
+@attr.s(auto_attribs=True)
+class ModelWithAnyJsonProperties:
+    """  """
+
+    additional_properties: Dict[
+        str, Union[ModelWithAnyJsonPropertiesAdditionalProperty, List[str], str, float, int, bool]
+    ] = attr.ib(init=False, factory=dict)
+
+    def to_dict(self) -> Dict[str, Any]:
+
+        field_dict: Dict[str, Any] = {}
+        for prop_name, prop in self.additional_properties.items():
+            if isinstance(prop, ModelWithAnyJsonPropertiesAdditionalProperty):
+                field_dict[prop_name] = prop.to_dict()
+
+            elif isinstance(prop, list):
+                field_dict[prop_name] = prop
+
+            elif isinstance(prop, str):
+                field_dict[prop_name] = prop
+            elif isinstance(prop, float):
+                field_dict[prop_name] = prop
+            elif isinstance(prop, int):
+                field_dict[prop_name] = prop
+            else:
+                field_dict[prop_name] = prop
+
+        field_dict.update({})
+
+        return field_dict
+
+    @staticmethod
+    def from_dict(src_dict: Dict[str, Any]) -> "ModelWithAnyJsonProperties":
+        d = src_dict.copy()
+        model_with_any_json_properties = ModelWithAnyJsonProperties()
+
+        additional_properties = {}
+        for prop_name, prop_dict in d.items():
+
+            def _parse_additional_property(
+                data: Any,
+            ) -> Union[ModelWithAnyJsonPropertiesAdditionalProperty, List[str], str, float, int, bool]:
+                data = None if isinstance(data, Unset) else data
+                additional_property: Union[
+                    ModelWithAnyJsonPropertiesAdditionalProperty, List[str], str, float, int, bool
+                ]
+                try:
+                    additional_property = ModelWithAnyJsonPropertiesAdditionalProperty.from_dict(data)
+
+                    return additional_property
+                except:  # noqa: E722
+                    pass
+                try:
+                    additional_property = cast(List[str], data)
+
+                    return additional_property
+                except:  # noqa: E722
+                    pass
+                return cast(str, prop_dict)
+                return cast(float, prop_dict)
+                return cast(int, prop_dict)
+                return cast(bool, prop_dict)
```

Comment:
It would do this before, just without the `cast`s. It's because in [`union_property.pyi`](https://github.com/triaxtec/openapi-python-client/blob/bbc550ecbaab08052e7a6b11dc42d4e7803b33b2/openapi_python_client/templates/property_templates/union_property.pyi#L18) we loop through each property and if it doesn't have a template we just return it.

What is here is ugly, yes, but it works. However, I do think there is an issue if we declare a templated type before a primitive type - then the primitive type would cause it to return early. Something , like 

```python

try:
    additional_property = cast(List[str], data)

    return additional_property
except:  # noqa: E722
    pass
return cast(str, prop_dict)
try:
    additional_property = ModelWithAnyJsonPropertiesAdditionalProperty.from_dict(data)

    return additional_property
except:  # noqa: E722
    pass
return cast(float, data)
return cast(int, data)
return cast(bool, data)
```

So I think we need to loop over templated properties first, then untemplated properties. Also we'll want to figure out lists, as casting it to a `List` will not error so that breaks the try/catch logic.

----

While responding to this comment, I realized that these should be using `cast(<primitive type>, data)` not `cast(<primitive type>, prop_dict)` - I added commits to this branch to change that

Reply:
I'll try looking at this soon as I have time